mlrun/api/db/sqldb/db.py

```diff
@@ -3818,3 +3820,59 @@ def set_run_notifications(
                 project=project,
             )
         self._commit(session, [run], ignore=True)
+
+    def delete_table_records(
+        self,
+        session: Session,
+        table: typing.Type[Base],
+        raise_on_not_exists=True,
+    ):
+        """Delete all records from a table
+
+        :param session: SQLAlchemy session
+        :param table: the table class
+        :param raise_on_not_exists: raise an error if the table does not exist
+        """
+        return self.delete_table_records_by_name(
+            session, table.__tablename__, raise_on_not_exists
+        )
+
+    def delete_table_records_by_name(
+        self,
+        session: Session,
+        table_name: str,
+        raise_on_not_exists=True,
+    ):
+        """
+        Delete a table by its name
+
+        :param session: SQLAlchemy session
+        :param table_name: table name
+        :param raise_on_not_exists: raise an error if the table does not exist
+        """
+        if self._is_table_exists(session, table_name):
+            truncate_statement = sqlalchemy.text(f"DELETE FROM {table_name}")
```

Comment:
sql injection alert
```suggestion
            truncate_statement = sqlalchemy.text(f"DELETE FROM :table_name").params(table_name=table_name)
```

Reply:
@liranbg the DELETE statement doesn't support a parameter placeholder.
I will add regex sanitation to the table name, and `_is_table_exists` provides another validation on the table name.